pkg/runners/zero/zero.go

```diff
@@ -272,13 +272,14 @@ func (runner *ZeroRunner) RunFor(steps uint64) error {
 // until the checkUsedCells doesn't return any error.
 // Since this vm always finishes the run of the program at the number of steps that is a power of two in the proof mode,
 // there is no need to run additional steps before the loop.
-func (runner *ZeroRunner) EndRun() {
+func (runner *ZeroRunner) EndRun() error {
 	for runner.checkUsedCells() != nil {
 		pow2Steps := utils.NextPowerOfTwo(runner.vm.Step + 1)
 		if err := runner.RunFor(pow2Steps); err != nil {
-			panic(err)
+			return err
```

Comment:
is it required not to panic here?

Reply:
cleaner imo, all of the steps after the execution are handled with an error, so I thought why panic here

Reply:
Makes sense